### PR TITLE
use cached dotnet in appveyor

### PR DESCRIPTION
**Pull request by TravisEz13**
Also, update build.psm to not always re-install dotnet cli.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -1192,8 +1192,36 @@ function Start-PSBootstrap {
...
-        $DotnetArguments = @{ Channel=$Channel; Version=$Version; NoSudo=$NoSudo }
-        Install-Dotnet @DotnetArguments
+        $dotNetExists = precheck 'dotnet' $null
+        $dotNetVersion = [string]::Empty
+        if($dotNetExists)
+        {
+            $dotNetVersion = (dotnet --version)
+        }
+
+        if(!$dotNetExists -or $dotNetVersion -ne '2.0.0-preview2-006502' -or $Force.IsPresent)
```
Comment (SteveL-MSFT) [inline build.psm1]: Use `$dotnetCLIRequiredVersion`

**Review by SteveL-MSFT — DISMISSED** [review]

File: build.psm1 [hunk 2]
```diff
@@ -1192,8 +1192,36 @@ function Start-PSBootstrap {
...
+        {
+            if($Force.IsPresent)
+            {
+                log "Installing dotnet due to -Force."
+
```
Comment (daxian-dbw) [inline build.psm1]: Remove extra new line.

File: build.psm1 [hunk 3]
```diff
@@ -1192,8 +1192,36 @@ function Start-PSBootstrap {
...
+                log "dotnet out of date ($dotNetVersion).  Updating dotnet."
+            }
+
+            $DotnetArguments = @{ Channel=$Channel; Version=$Version; NoSudo=$NoSudo }
+            Install-Dotnet @DotnetArguments
+        }
+        else
+        {
+            log "dotnet is already installed.  Skipping installation."
+        }
```
Comment (daxian-dbw) [inline build.psm1]: The existing scripting style is to not put the opening curly brace in a new line, for example:
```
if ($IsWindows) {
...
So I think it's better to follow the same style.
```

File: tools/appveyor.psm1 [hunk 4]
```diff
@@ -225,7 +231,7 @@ function Invoke-AppVeyorInstall
     }
 
     Set-BuildVariable -Name TestPassed -Value False
-    Start-PSBootstrap -Force
+    Start-PSBootstrap
```
Comment (daxian-dbw) [inline tools/appveyor.psm1]: I noticed that `Start-PSBootStrap` is defined with `ConfirmImpact="High"` and there are two places it asks for confirmation -- installing chocolatey and updating the machine Path environment variable. So will removing `-Force` cause any problems?

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**